specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/environmentContainer.json

```diff
@@ -0,0 +1,156 @@
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/environments": {
+      "get": {
+        "tags": [
+          "EnvironmentContainer"
+        ],
+        "operationId": "EnvironmentContainers_List",
+        "x-ms-examples": {
+          "List Environment Specifications.": {
+            "$ref": "./examples/listEnvironmentContainers.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/PaginationParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentContainerResourceArmPaginatedResult"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {},
+  "definitions": {
+    "EnvironmentContainer": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The asset container description text.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "The asset tag dictionary. Tags can be added, removed, and updated.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "latestVersions": {
+          "description": "Latest Versions for all assets",
+          "type": "array",
+          "items": {
+            "$ref": "environmentSpecification.json#/definitions/EnvironmentSpecificationVersionResource"
+          }
+        }
+      }
+    },
+    "EnvironmentContainerResource": {
```

Comment:
Is it possible to make this model inherit from some predefined resource base model in `common-types/resource-management/v2/types.json#/definitions`?

Reply:
yup, great catch! looks like I can use types.json's Resource definition. I can't believe I never saw that 👍

Reply:
ok it looks like there's some conflict between types.json and machineLearningServices.json definitions and I can't figure out how to fix it without breaking other parts of the pipeline. So instead I'll create a definition in machineLearningServices.json and use that instead